src/streaming/utils/CapabilitiesFilter.js

```diff
@@ -162,19 +162,97 @@ function CapabilitiesFilter() {
+                cfg.colorGamut = null;
+            }
+            else if (prop.schemeIdUri == Constants.COLOUR_PRIMARIES_SCHEME_ID_URI) {
+                cfg.isSupported = false;
+            }
+
+            // translate TransferCharacteristics signaling into capability queries
+            if (prop.schemeIdUri == Constants.TRANSFER_CHARACTERISTICS_SCHEME_ID_URI && ['1', '6', '13', '14', '15'].includes(prop.value.toString())) {
+                cfg.transferFunction = 'srgb';
+            }
+            else if (prop.schemeIdUri == Constants.TRANSFER_CHARACTERISTICS_SCHEME_ID_URI && ['16'].includes(prop.value.toString())) {
+                cfg.transferFunction = 'pq';
+            }
+            else if (prop.schemeIdUri == Constants.TRANSFER_CHARACTERISTICS_SCHEME_ID_URI && ['18'].includes(prop.value.toString())) {
+                cfg.transferFunction = 'hlg';
+            }
+            else if (prop.schemeIdUri == Constants.TRANSFER_CHARACTERISTICS_SCHEME_ID_URI && ['2'].includes(prop.value.toString())) {
+                cfg.transferFunction = null;
+            }
+            else if (prop.schemeIdUri == Constants.TRANSFER_CHARACTERISTICS_SCHEME_ID_URI) {
+                cfg.isSupported = false;
+            }
+        }
+
+        return cfg;
+    }
+
+    function _convertHDRMetadataFormatToConfig(representation) {
+        let cfg = {
+            isSupported: true
         };
+
+        for (const prop of representation.EssentialProperty || []) {
+            // translate hdrMetadataType signaling into capability queries
+            if (prop.schemeIdUri == Constants.HDR_METADATA_FORMAT_SCHEME_ID_URI && prop.value == 'ST2094-10') {
+                cfg.hdrMetadataType = 'smpteSt2094-10';
+            }
+            else if (prop.schemeIdUri == Constants.HDR_METADATA_FORMAT_SCHEME_ID_URI && prop.value == 'SL-HDR2') {
+                cfg.hdrMetadataType = 'slhdr2'; // Note: This is not specified by W3C
+            }
+            else if (prop.schemeIdUri == Constants.HDR_METADATA_FORMAT_SCHEME_ID_URI && prop.value == 'ST2094-40') {
+                cfg.hdrMetadataType = 'smpteSt2094-40';
+            }
+            else if (prop.schemeIdUri == Constants.HDR_METADATA_FORMAT_SCHEME_ID_URI) {
+                cfg.isSupported = false;
+            }
+        }
+
+        return cfg;
+    }
+
+    function _createVideoConfiguration(rep, codec) {
+        let config = {
+            codec: codec,
+            width: rep.width || null,
+            height: rep.height || null,
+            framerate: rep.frameRate || null,
+            bitrate: rep.bandwidth || null
+        }
+        if (settings.get().streaming.capabilities.filterVideoColorimetryEssentialProperties) {
+            Object.assign(config, _convertHDRColorimetryToConfig(rep));
+        }
+        let supportedFlag = config.isSupported;
+        if (settings.get().streaming.capabilities.filterHDRMetadataFormatEssentialProperties) {
```

Comment:
I suggest to use two different flags here:
* `metadataFormatSupported`
* `colorimetrySupported`